src/test/librbd/crypto/luks/test_mock_FlattenRequest.cc

```diff
@@ -41,7 +41,7 @@ struct TestMockCryptoLuksFlattenRequest : public TestMockFixture {
   typedef FlattenRequest<MockTestImageCtx> MockFlattenRequest;
 
   const size_t OBJECT_SIZE = 4 * 1024 * 1024;
-  const uint64_t DATA_OFFSET = MockCryptoInterface::DATA_OFFSET;
+  const uint64_t DATA_OFFSET = MockCryptoInterface::MOCK_CRYPTO_INTERFACE_DATA_OFFSET;
```

Comment:
the BOOST_ASIO_HAS_IO_URING define causes boost/asio.hpp to include this header:
```
In file included from /usr/include/liburing/io_uring.h:11,
                 from /usr/include/liburing.h:25,
                 from /home/cbodley/ceph/build/boost/include/boost/asio/detail/io_uring_operation.hpp:22,
                 from /home/cbodley/ceph/build/boost/include/boost/asio/detail/io_uring_descriptor_read_at_op.hpp:28,
                 from /home/cbodley/ceph/build/boost/include/boost/asio/detail/io_uring_descriptor_service.hpp:28,
                 from /home/cbodley/ceph/build/boost/include/boost/asio/detail/io_uring_file_service.hpp:26,
                 from /home/cbodley/ceph/build/boost/include/boost/asio/basic_file.hpp:39,
                 from /home/cbodley/ceph/build/boost/include/boost/asio.hpp:28,
                 from /home/cbodley/ceph/src/include/neorados/RADOS.hpp:27,
                 from /home/cbodley/ceph/src/test/librbd/mock/MockImageCtx.cc:4:
/home/cbodley/ceph/src/test/librbd/mock/crypto/MockCryptoInterface.h:16:25: error: expected unqualified-id before numeric constant
   16 |   static const uint64_t BLOCK_SIZE = 4096;
      |                         ^~~~~~~~~~
```
cc @adamemerson we might consider changing neorados/RADOS.hpp to use more granular includes of asio headers to avoid this dependency on liburing (and to reduce compilation times in general)

Reply:
> we might consider changing neorados/RADOS.hpp to use more granular includes of asio headers

i added some more commits for this, along with a SQUASHME commit that reverts the changes to this librbd test